onnxruntime/core/framework/provider_bridge_ort.cc

```diff
@@ -1217,3 +1235,117 @@ ORT_API_STATUS_IMPL(OrtApis::SessionOptionsAppendExecutionProvider_CUDA, _In_ Or
+  (*out)->device_id = 0;
+  (*out)->has_user_compute_stream = 0;
+  (*out)->user_compute_stream = nullptr;
+  (*out)->trt_max_partition_iterations = 1000;
+  (*out)->trt_min_subgraph_size = 1;
+  (*out)->trt_max_workspace_size = 1 << 30;
+  (*out)->trt_fp16_enable = false;
+  (*out)->trt_int8_enable = false;
+  (*out)->trt_int8_calibration_table_name = nullptr;
+  (*out)->trt_int8_use_native_calibration_table = false;
+  (*out)->trt_dla_enable = false;
+  (*out)->trt_dla_core = false;
+  (*out)->trt_dump_subgraphs = false;
+  (*out)->trt_engine_cache_enable= false;
+  (*out)->trt_engine_cache_path = nullptr;
+  (*out)->trt_engine_decryption_enable = false;
+  (*out)->trt_engine_decryption_lib_path = nullptr;
+  (*out)->trt_force_sequential_engine_build = false;
+  return nullptr;
+#else
+  ORT_UNUSED_PARAMETER(out);
+  return CreateStatus(ORT_FAIL, "TensorRT execution provider is not enabled in this build.");
+#endif
+  API_IMPL_END
+}
+
+ORT_API_STATUS_IMPL(OrtApis::UpdateTensorRTProviderOptions,
+                    _Inout_ OrtTensorRTProviderOptionsV2* tensorrt_options,
+                    _In_reads_(num_keys) const char* const* provider_options_keys,
+                    _In_reads_(num_keys) const char* const* provider_options_values,
+                    size_t num_keys) {
+  API_IMPL_BEGIN
+#ifdef USE_TENSORRT
+  onnxruntime::ProviderOptions provider_options_map;
+  for (size_t i = 0; i != num_keys; ++i) {
+    if (provider_options_keys[i] == nullptr || provider_options_keys[i][0] == '\0' ||
+        provider_options_values[i] == nullptr || provider_options_values[i][0] == '\0') {
+      return OrtApis::CreateStatus(ORT_INVALID_ARGUMENT, "key/value cannot be empty");
+    }
+
+    provider_options_map[provider_options_keys[i]] = provider_options_values[i];
+  }
+
+  onnxruntime::UpdateProviderInfo_Tensorrt(reinterpret_cast<OrtTensorRTProviderOptions*>(tensorrt_options),
+                                           reinterpret_cast<const onnxruntime::ProviderOptions&>(provider_options_map));
+  return nullptr;
+#else
+  ORT_UNUSED_PARAMETER(tensorrt_options);
+  ORT_UNUSED_PARAMETER(provider_options_keys);
+  ORT_UNUSED_PARAMETER(provider_options_values);
+  ORT_UNUSED_PARAMETER(num_keys);
+  return CreateStatus(ORT_FAIL, "TensorRT execution provider is not enabled in this build.");
+#endif
+  API_IMPL_END
+}
+
+ORT_API_STATUS_IMPL(OrtApis::GetTensorRTProviderOptionsAsString, _In_ const OrtTensorRTProviderOptionsV2* tensorrt_options, _Inout_ OrtAllocator* allocator,
+                    _Outptr_ char** ptr) {
+  API_IMPL_BEGIN
+#ifdef USE_TENSORRT
+  onnxruntime::ProviderOptions options = onnxruntime::GetProviderInfo_Tensorrt(reinterpret_cast<const OrtTensorRTProviderOptions*>(tensorrt_options));
+  onnxruntime::ProviderOptions::iterator it = options.begin();
+  std::string options_str = "";
+
+  while (it != options.end()) {
+    if (options_str == "") {
+      options_str = it->first + "=" + it->second;
```

Comment:
Prefer .append() or +=

Reply:
changed to use append().

Reply:
Using the '+' operator creates a new copy of the string. When you're appending to a string, it's better to either use += (same as append) or ostringstream. It is not a big deal here since this is not in the performance path. But just for your knowledge, this would be something like 
options_str += it->first;
options_str += "=";
options_str += it->second;

Or
std::ostringstream ostr;
ostr << it->first << "=" << it->second;
options_str = ostr.str();

Reply:
Thanks for the explanation. Have changed the code here to use string append "+=" rather than string copy '+'.